cylc/uiserver/resolvers.py

```diff
@@ -172,80 +166,72 @@ def func(schema_key, schema_value):
         else:
             api_opts[opt] = value
-    return SimpleNamespace(**api_opts)
-
-
-def _clean(tokens, opts):
-    """Run Cylc Clean using `cylc.flow.workflow_files` api.
-    """
-    try:
-        init_clean(tokens.pop('workflow'), opts)
-    except ServiceFileError as exc:
-        return str(exc).split('\n')[0]
-    else:
-        return 'Workflow cleaned'
+    return schema(**api_opts)
 
 
 class Services:
     """Cylc services provided by the UI Server."""
 
     @staticmethod
-    def _error(message):
+    def _error(message: Union[Exception, str]):
         """Format error case response."""
         return [
             False,
             str(message)
         ]
 
     @staticmethod
-    def _return(message):
+    def _return(message: str):
         """Format success case response."""
         return [
             True,
             message
         ]
 
     @classmethod
-    async def clean(cls, workflows, args, workflows_mgr, executor, log):
+    async def clean(
+        cls,
+        workflows: Iterable['Tokens'],
+        args: dict,
+        workflows_mgr: 'WorkflowsManager',
+        executor: 'Executor',
+        log: 'Logger'
+    ):
         """Calls `init_clean`"""
         # Convert Schema options → cylc.flow.workflow_files.init_clean opts:
-        try:
-            opts = _schema_opts_to_api_opts(args, schema=CLEAN)
-        except Exception as exc:
-            return cls._error(exc)
+        opts = _schema_opts_to_api_opts(args, schema=CleanOptions)
         # Hard set remote timeout.
         opts.remote_timeout = "600"
 
         # clean each requested flow
         for tokens in workflows:
-            clean_func = partial(_clean, tokens, opts)
+            log.info(f'Cleaning {tokens}')
+            workflow = tokens.pop('workflow')
             try:
-                log.info(f'Cleaning {tokens}')
-                future = await asyncio.wrap_future(executor.submit(clean_func))
+                await asyncio.wrap_future(
+                    executor.submit(init_clean, workflow, opts)
+                )
             except Exception as exc:
-                log.exception(exc)
-                return cls._error(exc)
-            else:
-                if future == WORKFLOW_RUNNING_MSG:
-                    log.error(ServiceFileError(WORKFLOW_RUNNING_MSG))
-                    return cls._error(WORKFLOW_RUNNING_MSG)
+                if isinstance(exc, ServiceFileError):  # Expected error
+                    # The "workflow still running" msg is very long
+                    msg = str(exc).split('\n')[0]
+                elif isinstance(exc, WorkflowFilesError):  # Expected error
+                    msg = str(exc)
+                else:  # Unexpected error
+                    msg = f"{type(exc).__name__}: {exc}"
+                    log.exception(msg)
+                return cls._error(msg)
 
         # trigger a re-scan
-        await workflows_mgr.update()
+        await workflows_mgr.scan()
```

Comment:
As Oliver said in #368 
> This is due to a change in the scanning logic, the `.scan` interface requests rather than performs a scan preventing concurrent scan operations from fighting over the same workflows.